File: algorithms/easy/minimum_common_value.ex
defmodule Solution do
  @spec get_common(nums1 :: [integer], nums2 :: [integer]) :: integer
  def get_common([], _), do: -1
  def get_common(_, []), do: -1

  def get_common([n | ns], [m | ms]) do
    cond do
      n == m -> n
      n < m -> get_common(ns, [m | ms])
      true -> get_common([n | ns], ms)
    end
  end
end

File: algorithms/easy/minimum_common_value.cpp
class Solution {
public:
  int getCommon(vector<int> &nums1, vector<int> &nums2) {
    int l = 0, r = 0;
    while (l < nums1.size() && r < nums2.size()) {
      if (nums1[l] == nums2[r])
        return nums1[l];
      if (nums1[l] < nums2[r]) {
        l++;
      } else {
        r++;
      }
    }
    return -1;
  }
};
